### PR TITLE
Add zsh completion for 'docker {service ls,stack services} --format'

### PR DESCRIPTION
**Pull request by sdurrheimer**
ref. #28199

This can be planned for `1.14.0`.

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -1845,6 +1845,7 @@ __docker_service_subcommand() {
             _arguments $(__docker_arguments) \
                 $opts_help \
                 "($help)*"{-f=,--filter=}"[Filter output based on conditions provided]:filter:->filter-options" \
+                "($help)--format=[Pretty-print services using a Go template]:template: " \
```
Comment (cpuguy83) [inline contrib/completion/zsh/_docker]: Curious what the difference in format for, e.g., `-q` and this is about?

Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: Not sure to properly understand the question :thinking:

Comment (cpuguy83) [inline contrib/completion/zsh/_docker]: Sorry, format was not the right word given that's the flag here... but looks like flags with a single representation use a different syntax than ones with multiple representations (e.g. `-q` and `--quiet` vs simply `--format`).

Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: Yes that's a zsh completion syntax thing. Similar options must be set in the following format `{option1,option2,...}`.
The `=` sign means the value can be set like that: `--option=value` or `--option value`.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**